CONTRIBUTING.md

```diff
@@ -0,0 +1,136 @@
+7. If this branch fixes an issue, use github's `fixes: #issue_number` when
+   writing your commit message to link the issue to your pr.
+8. Push to your fork and create a pull request.
+9. Ensure that the test suite passes CI and make any necessary changes to your
+   branch to bring it to green.
+
+We aim to review pull requests at our internal Ackama fortnightly ruby guild
+meeting. Once we look at your pull request, we may give you feedback. For
+instance, we may suggest some changes to make to your code to fit within the
+project style or discuss alternate ways of addressing the issue in question.
+Assuming we're happy with everything, we'll then bring your changes into main.
+Now you're a contributor!
+
+## Code of Conduct
+
+If this is your first time contributing, please read the [Code of Conduct]. We
+want to create a space in which everyone is allowed to contribute, and we
+enforce the policies outline in this document.
+
+[code of conduct]:
+  https://github.com/ackama/rails-template/blob/main/CODE_OF_CONDUCT.md
+
+## Setting up your environment
+
+1. Install
+   [the most recent ruby version](https://www.ruby-lang.org/en/downloads/) with
+   the ruby version manager of your choice
+2. Run `gem install rails` to get the most recent rails release
+3. Install the node version defined in the `.node-version` file with the node
+   version manager of your choice
+4. Install yarn v1
+
+To run tests you'll also need to install
+
+- [PostgreSQL](https://www.postgresql.org/)
+- [Chromedriver](https://chromedriver.chromium.org/)
+
+## Architecture
+
+This project works by hooking into the standard Rails
+[application templates](https://guides.rubyonrails.org/rails_application_templates.html)
+system, with some caveats. The entry point is the
+[template.rb](https://github.com/ackama/rails-template/blob/main/template.rb)
+file in the root of this repository.
+
+Normally, Rails only allows a single file to be specified as an application
+template (i.e. using the `-m <URL>` option). To work around this limitation, the
+first step this template performs is a `git clone` of the
+`ackama/rails-template` repository to a local temporary directory.
+
+This temporary directory is then added to the `source_paths` of the Rails
+generator system, allowing all of its ERb templates and files to be referenced
+when the application template script is evaluated.
+
+Rails generators are very lightly documented; what you'll find is that most of
+the heavy lifting is done by [Thor](http://whatisthor.com/). Thor is a tool that
+allows you to easily perform command line utilities. The most common methods
+used by this template are Thor's `copy_file`, `template`, and `gsub_file`. You
+can dig into the well-organized and well-documented
+[Thor source code](https://github.com/erikhuda/thor) to learn more. If any file
+finishes with `.tt`, Thor considers it to be a template and places it in the
+destination without the extension `.tt`.
+
+## Tests
```

Comment:
This section feels unfinished to me - notably, I don't think it correctly explains that:

1. we have a collection of configs that are run by CI
2. these live in `ci/configs`
3. `ci/bin/build-and-test` can be used to run _one_ of these configs